### PR TITLE
speedup build times for incremental builds

### PR DESCRIPTION
**Pull request by michaelknoch**
we call cmake from cmake, due to some gradle limitations. Make sure we call **cmake configure** from **cmake configure** and **cmake build** from **cmake build**.

If you already built the player, every incremental build will be way faster because cmake will only rebuild sources which might have changed or skip build completely if all sourcefiles are untouched

### CODE REVIEW

File: BuildSwiftProjectConfig.cmake [hunk 1]
```diff
@@ -20,9 +20,13 @@ set(LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/src/main/jniLibs/${AND
 set(NEVER_EXISTING_OUTPUT ./nop)
 
 function(build_swift_project)
+    execute_process(
+        COMMAND bash -c "LIBRARY_OUTPUT_DIRECTORY=${LIBRARY_OUTPUT_DIRECTORY} ANDROID_ABI=${ANDROID_ABI} CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} ${SWIFT_TOOLCHAIN_ROOT}/swift-build.sh -configure ${PROJECT_DIRECTORY}"
```
Comment (ephemer) [inline BuildSwiftProjectConfig.cmake]: is this supposed to be `-configure` or `--configure`?

**Comment by michaelknoch** [comment]
ready for final review @ephemer :)

File: swift-build.sh [hunk 2]
```diff
@@ -58,4 +56,26 @@ build() {
     echo "Finished compiling ${CMAKE_BUILD_TYPE} for ${ANDROID_ABI}"
 }
 
+readonly SCRIPT_ROOT=$(cd $(dirname $0); echo -n $PWD) # path of this file
+readonly SWIFT_ANDROID_TOOLCHAIN_PATH="${SWIFT_ANDROID_TOOLCHAIN_PATH:-$SCRIPT_ROOT}"
+
+for LAST_ARGUMENT in $@; do :; done
+readonly PROJECT_DIRECTORY=${LAST_ARGUMENT:-$PWD}
```
Comment (ephemer) [inline swift-build.sh]: Can you check whether we ever get into the fallback case here? Bash has some weird rules with this fallback syntax

File: swift-build.sh [hunk 3]
```diff
@@ -46,9 +34,19 @@ build() {
...
+        -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE \
+        -S ${PROJECT_DIRECTORY} \
+        -B ${BUILD_DIR}
+    
+    echo "Finished configure ${CMAKE_BUILD_TYPE} for ${ANDROID_ABI}"
+}
+
+build() {
+    # reconfigure when build dir does not exist empty
+    [[ -d ${BUILD_DIR} ]] || configure
```
Comment (ephemer) [inline swift-build.sh]: Nice. How’d you get this working in the end?

Comment (michaelknoch) [reply swift-build.sh]: We still tried to cmake —build . (In current directory) without cding into. Changing it to cmake —build $BUILD_DIR did the trick. @ephemer

File: swift-build.sh (hunk 2 above)
Comment (michaelknoch) [reply swift-build.sh]: jep, works like charm

**Pull request merged**